debug/org.eclipse.debug.core/core/org/eclipse/debug/core/DebugPlugin.java

```diff
@@ -727,22 +734,49 @@ public void stop(BundleContext context) throws Exception {
 	public void start(BundleContext context) throws Exception {
 		super.start(context);
 		new DebugOptions(context);
-		ResourcesPlugin.getWorkspace().addSaveParticipant(getUniqueIdentifier(),
-				new ISaveParticipant() {
-					@Override
-					public void saving(ISaveContext saveContext) throws CoreException {
-						if (fExpressionManager != null) {
-							fExpressionManager.storeWatchExpressions();
+
+		fWorkspaceServiceTracker = new ServiceTracker<>(context, IWorkspace.class, null) {
+
+			@Override
+			public IWorkspace addingService(ServiceReference<IWorkspace> reference) {
+				IWorkspace workspace = super.addingService(reference);
+				try {
+					workspace.addSaveParticipant(getUniqueIdentifier(), new ISaveParticipant() {
```

Comment:
workspace might be null at this point...

Reply:
I added a guard now.